src/Command/MauticLanguagePackerCommand.php

```diff
@@ -0,0 +1,130 @@
+            'languages',
+            'l',
+            InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+            'Process languages. For e.g. bin/console '.self::NAME.' -l es -l en.'
+        );
+        $this->addOption(
+            'upload-package',
+            'u',
+            InputOption::VALUE_NONE,
+            'Upload the package to AWS S3. For e.g. bin/console '.self::NAME.' -u.'
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io     = new SymfonyStyle($input, $output);
+        $logger = $this->getConsoleLogger($output);
+
+        $packagesDir     = $this->parameterBag->get('mlp.packages.dir');
+        $translationsDir = $this->parameterBag->get('mlp.translations.dir');
+
+        $skipLanguages = $input->getOption('skip-languages');
+        $languages     = $input->getOption('languages');
+        $uploadPackage = $input->getOption('upload-package');
+
+        // Remove any previous pulls and rebuild the translations folder
+        $this->filesystem->remove($translationsDir);
+        $this->filesystem->mkdir($translationsDir);
+
+        // Fetch the project resources now and store them locally
+        $resourceDTO   = new ResourceDTO($translationsDir, $skipLanguages, $languages);
+        $resourceError = $this->resourcesService->processAllResources($resourceDTO, $logger);
+
+        if ($resourceError) {
+            $io->error('Encountered error during fetching all resources.');
+
+            return Command::FAILURE;
+        }
+
+        // Now we start building our ZIP archives
+        $this->filesystem->mkdir($packagesDir);
+
+        // Add a folder for our current build
+        $timestamp            = (new \DateTime())->format('YmdHis');
+        $packagesTimestampDir = $packagesDir.'/'.$timestamp;
+        $this->filesystem->mkdir($packagesTimestampDir);
+
+        // Compile our data to forward to mautic.org and build the ZIP packages
+        $packageDTO = new PackageDTO($translationsDir, $skipLanguages, $packagesTimestampDir);
+        $buildError = $this->buildPackageService->build($packageDTO, $logger);
+
+        if ($buildError) {
+            $io->warning('Created language packages for Mautic with some errors.');
+        } else {
+            $io->success('Successfully created language packages for Mautic!');
+        }
+
+        // If instructed, upload the packages
+        if ($uploadPackage) {
+            $io->info('Starting package upload to AWS S3.');
+            $uploadPackageDTO = new UploadPackageDTO($packagesTimestampDir);
+            $uploadError      = $this->uploadPackageService->uploadPackage($uploadPackageDTO, $logger);
+
+            if ($uploadError) {
+                $io->error('Encountered error during language packages upload to AWS S3.');
+            } else {
+                $io->success('Successfully uploaded language packages to AWS S3!');
+            }
+        }
```

Comment:
```suggestion
        if ($uploadPackage) {
            $io->info('Starting package upload to AWS S3.');
            $this->uploadPackageService->uploadPackage(new UploadPackageDTO($packagesTimestampDir), $logger);
            $io->info('Package upload to AWS S3 finished.');
        }
```
Let's move the the messages about the partial success to the service. This way the method `uploadPackage()` can return `void`. Btw. prefer returning `boolean` to `int` in general.

Reply:
The comment above also applies to the following lines:
https://github.com/mautic/language-packer/blob/5718b0c74deabd441e2e101ae1da9b0c54c538de/src/Command/MauticLanguagePackerCommand.php#L96-L103